gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageGrpcIntegrationTest.java

```diff
@@ -37,31 +44,58 @@ public class GoogleCloudStorageGrpcIntegrationTest {
 
   private static final String BUCKET_NAME = BUCKET_HELPER.getUniqueBucketName("shared");
 
-  private static GoogleCloudStorage createGoogleCloudStorage() throws IOException {
+  private final boolean tdEnabled;
+
+  @Parameters
+  // We want to test this entire class with both gRPC-LB and Traffic Director
+  public static Iterable<Boolean> tdEnabled() {
+    return Arrays.asList(false, true);
+  }
+
+  public GoogleCloudStorageGrpcIntegrationTest(boolean tdEnabled) {
+    this.tdEnabled = tdEnabled;
+  }
+
+  private GoogleCloudStorageOptions.Builder configureOptions() {
+    GoogleCloudStorageOptions.Builder optionsBuilder =
+        GoogleCloudStorageTestHelper.getStandardOptionBuilder()
+            .setGrpcEnabled(true)
+            .setTrafficDirectorEnabled(this.tdEnabled);
+    boolean runningTestsFromGCP = Boolean.parseBoolean(System.getenv("RUNNING_TESTS_FROM_GCP_VM"));
```

Comment:
Instead of accepting a boolean and hardcoding the endpoint, can we accept endpoint as a configuration with a default value?